generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -81,27 +73,17 @@ private boolean mustBeNull(FieldSpec fieldSpec) {
         return Collections.singletonList(new CannedValuesFieldValueSource(whitelist.collect(Collectors.toList())));
     }
 
-    private Stream<FieldSpec> getNotNullSetRestrictionFilterOnMustContainRestriction(MustContainRestriction restriction) {
-        return restriction.getRequiredObjects()
-            .stream()
-            .filter(o -> o.getSetRestrictions() != null
-                && o.getSetRestrictions().getWhitelist() != null);
-    }
-
     private boolean mayBeNull(FieldSpec fieldSpec) {
         return fieldSpec.getNullRestrictions() == null;
     }
 
     private List<FieldValueSource> getMustContainRestrictionSources(FieldSpec fieldSpec) {
         Set<FieldSpec> mustContainRestrictionFieldSpecs = fieldSpec.getMustContainRestriction().getRequiredObjects();
-        if (mustContainRestrictionFieldSpecs.size() > 1) {
-            mustContainRestrictionFieldSpecs = mustContainRestrictionReducer.getReducedMustContainRestriction(fieldSpec);
-        }
 
         return FlatMappingSpliterator.flatMap(mustContainRestrictionFieldSpecs.stream()
             .map(this::getFieldValueSources),
             List::stream)
-            .filter(x->!x.equals(nullOnlySource))
+            .distinct()
```

Comment:
This means that null could be emitted, albeit only once, is this correct?

Reply:
this is as desired because now the must contain has all the information needed for generation. It no longer generates a  `null` at a later point (from the root field spec).